internal/logic/logic.go

```diff
@@ -49,5 +63,167 @@ func (l htmlPdfServiceLogic) Ping(req *model.PingRequest) *respModel.Response {
+	fileBytes, err := ioutil.ReadAll(file)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrReadFileFail),
+			Data:    nil,
+		}
+	}
+	jb, err := l.htSvc.PDFPreparer(fileBytes)
+	if err != nil {
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileConversionFail),
+			Data:    nil,
+		}
+	}
+	u := uuid.NewString()
+	err = l.dsSvc.Set(u, jb, 0)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileStoreFail),
+			Data:    nil,
+		}
+	}
+	return &respModel.Response{
+		Status:  http.StatusCreated,
+		Message: "SUCCESS",
+		Data: map[string]interface{}{
+			"id": u,
+		},
+	}
+}
+
+func (l htmlPdfServiceLogic) Replace(id string, file io.Reader) *respModel.Response {
+	_, err := l.dsSvc.Get(id)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrKeyNotFound),
+			Data:    nil,
+		}
+	}
+	fileBytes, err := ioutil.ReadAll(file)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrReadFileFail),
+			Data:    nil,
+		}
+	}
+	jb, err := l.htSvc.PDFPreparer(fileBytes)
+	if err != nil {
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileConversionFail),
+			Data:    nil,
+		}
+	}
+
+	err = l.dsSvc.Set(id, jb, 0)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileStoreFail),
+			Data:    nil,
+		}
+	}
+	return &respModel.Response{
+		Status:  http.StatusCreated,
+		Message: "SUCCESS",
+		Data: map[string]interface{}{
+			"id": id,
+		},
+	}
+}
+
+func (l htmlPdfServiceLogic) HtmlToPdf(w io.Writer, req *model.GenerateReq) *respModel.Response {
+	var z map[string]interface{}
+	b, err := l.dsSvc.Get(req.Id)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFetchingFile),
+			Data:    nil,
+		}
+	}
+	err = json.NewDecoder(bytes.NewBuffer(b)).Decode(&z)
+	if err != nil {
+		log.Error("error unmarshaling JSON:" + err.Error())
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileParseFail),
+			Data:    nil,
+		}
+	}
+	for i, p := range z["Pages"].([]interface{}) {
+		page := p.(map[string]interface{})
+		buf, err := base64.StdEncoding.DecodeString(page["Base64PageData"].(string))
+		if err != nil {
+			log.Error("error decoding base 64 input on page" + fmt.Sprint(i) + err.Error())
+			return &respModel.Response{
+				Status:  http.StatusInternalServerError,
+				Message: codes.GetErr(codes.ErrReadFileFail),
```

Comment:
Where is file read happening here? Error code needs to be updated to reflect where the error has occurred.

Reply:
done